src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPExtendWordSelectionHandler.java

```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.selectionRange;
+
+import com.intellij.codeInsight.editorActions.ExtendWordSelectionHandler;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.containers.ContainerUtil;
+import com.redhat.devtools.lsp4ij.LSPFileSupport;
+import com.redhat.devtools.lsp4ij.LSPIJUtils;
+import com.redhat.devtools.lsp4ij.client.ExecuteLSPFeatureStatus;
+import com.redhat.devtools.lsp4ij.client.indexing.ProjectIndexingManager;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.SelectionRange;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static com.intellij.codeInsight.editorActions.ExtendWordSelectionHandlerBase.expandToWholeLinesWithBlanks;
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.isDoneNormally;
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.waitUntilDone;
+
+/**
+ * Implementation of the IDE's extendWordSelectionHandler EP for LSP4IJ files against textDocument/selectionRange.
+ */
+public class LSPExtendWordSelectionHandler implements ExtendWordSelectionHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LSPExtendWordSelectionHandler.class);
+
+    @Override
+    public boolean canSelect(@NotNull PsiElement element) {
+        if (!element.isValid()) {
+            return false;
+        }
+
+        PsiFile file = element.getContainingFile();
+        if ((file == null) || !file.isValid()) {
+            return false;
+        }
+
+        // Only if textDocument/selectionRange is supported for the file
+        return LSPFileSupport.getSupport(file).getSelectionRangeSupport() != null;
```

Comment:
This check is every time not null. You need to check if there is a language server which support selectionrange:

```java
// Check if the file can support the feature
return LanguageServiceAccessor.getInstance(project)
                .hasAny(file.getVirtualFile(), ls -> ls.getClientFeatures().getSelectionRangeFeature().isSelectionRangeSupported(file)));
```

Reply:
Please see what I've done in the latest commit. I looked for something that does what the new method `isSupported(PsiFile, Predicate<LSPClientFeatures>)` is doing but couldn't find anything quite like it. If something does exist, please let me know and I'll migrate to it; if not, perhaps that method should be moved somewhere more common for future needs? If so, where? `LanguageServerAccessor`? `LSP4IJUtils`? ???